### ISSUE TITLE
RNA-seq & Array issue

### ISSUE THREAD

**Issue by znelson999** [ncbi/egapx#10]
Hello,

I'm attempting to include RNA-seq data into my run and am getting an array error. The sequences are locally hosted and not in NCBI. Included is the error I'm getting and the yaml file being used.

[yamlFileUsed.txt](https://github.com/ncbi/egapx/files/15266751/yamlFileUsed.txt)

[EGAPissue.txt](https://github.com/ncbi/egapx/files/15266723/EGAPissue.txt)

The wording used in the documentation says that RNA-seq files have to end in 1 or 2; could that be causing my problems? If so is there a work around to be able to use data that is not on NCBI? Or could it just be an error in the way I've set up the required files?

Thank you for any assistance you can provide.

**Comment by pstrope** [comment]
Hi, Thank you for reporting. 

Can you please post the command you used? 

thanks,
Pooja

**Comment by pstrope** [comment]
Currently, the read files have to end in .1, .2  to be paired up. In future, we will make it more flexible. 

To be able to read the gz read files, edit the  `ui/assets/default_task_params.yaml`:
Under `star_wnode:`  is a `-star-params`
To the argument list, add this  `--readFilesCommand zcat`

**Comment by pstrope** [comment]
Also 
`python3 /project/cricket_gen/ZachN/EGAPx/egapx-main/ui/egapx.py /project/cricket_gen/ZachN/EGAPx/egapx-main/examples/input_D_farinae_small.yaml -o outdir` 
for your first egapx run.

**Comment by pstrope** [comment]
Hi, 
The best thing to do is to wait for our next version update. We are actively working on this issue that your brought up.
Currently, it's also having trouble reading gz read files.

We appreciate your testing and reporting. We'll reach out when the fix is ready. 

Pooja

**Comment by znelson999** [comment]
Very well, thank you for assistance.

**Comment by victzh** [comment]
@znelson999 can you supply me with a short sample of the FASTQ files you use - say first 100 lines of first 4 files in your list (after zcat'ting it of course)? By using something like
zcat /project/cricket_gen/ZachN/EGAPx/TenebrioMolTranscriptome/Tmol_RNA_transcriptome_lifestages/t3_ll_1_R1.fastq.gz | head -100 > t3_ll_1_R1.fastq.sample 
I want to ensure that our code works with the output from a real sequencing machine (which I assume it is) vs. processed reads from SRA.

**Comment by znelson999** [comment]
Hi @victzh 

Attached is a zip file with the portions you requested, let me know if this helps.

[fastqsamples.zip](https://github.com/ncbi/egapx/files/15325202/fastqsamples.zip)

**Comment by victzh** [comment]
@znelson999 thanks! It seems to me that t3_ll_1_R1.fastq.gz and t3_ll_1_R2.fastq.gz are parts of a paired run. Why the samples then match each other exactly? Shouldn't they be different ends of the same piece of RNA? I'm not a biologist, I'm a programmer, so pardon my ignorance.

**Comment by znelson999** [comment]
@victzh 

Thanks for pointing this out. I made a mistake when making those sample files. Attached are the appropriate zipped files.
[RealFastqFiles.zip](https://github.com/ncbi/egapx/files/15338277/RealFastqFiles.zip)

**Comment by victzh** [comment]
@znelson999 thanks, it helps a lot. I tried to run newer version with your data but it failed (so far) because the samples are too short. We need to fix this and will see what else fails.

**Comment by zilov** [comment]
Hello! I'm encountering same "index is out of range" error while running egapx with local RNA-Seq data. I've tried renaming my unzipped files to {prefix}.1, {prefix}.2, {prefix}.1.fastq, and {prefix}.2.fastq, but the error persists.

Here's the error message:
```
ERROR ~ index is out of range 0..-1 (index = 0)

 -- Check script 'nf/./subworkflows/ncbi/./rnaseq_short/star_wnode/main.nf' at line: 83 or see '/media/eternus1/data/vgp/glis_glis/users/zilov/annotation/egapx/egapx/glis_out_last/nextflow.log' file for more details
```
I installed egapx following the README instructions and am running it within the nextflow conda environment. Test run using the example data completed successfully.

Is there a way to troubleshoot this now? Should I wait for the next version of the tool, or is the problem likely on my side?

Input file: [input_yaml2.txt](https://github.com/ncbi/egapx/files/15456253/input_yaml2.txt)
